wait/metrics.go

```diff
@@ -37,14 +42,46 @@ func getCounter(url string, family string, labelKey string, labelValue string) (
 	}
 	for _, f := range families {
 		if f.GetName() == family {
+			metricType := f.GetType()
+			// metric without labels
+			if len(f.GetMetric()) == 1 && len(expectedLabels) == 0 {
+				return getValue(metricType, f.GetMetric()[0])
+			}
+
+		metricSearch:
 			for _, m := range f.GetMetric() {
-				for _, l := range m.GetLabel() {
-					if l.GetName() == labelKey && l.GetValue() == labelValue {
-						return m.GetCounter().GetValue(), nil
+				metricLabels := m.GetLabel()
+				if len(metricLabels) != len(expectedLabels)/2 {
+					continue
+				}
+				for i := 0; i < len(expectedLabels); {
+					labelFound := false
+					for _, l := range metricLabels {
+						if l.GetName() == expectedLabels[i] && l.GetValue() == expectedLabels[i+1] {
+							labelFound = true
+						}
 					}
+					if !labelFound {
+						continue metricSearch
+					}
+					i += 2
 				}
+				return getValue(metricType, m)
 			}
 		}
 	}
-	return -1, nil
+	return -1, fmt.Errorf("Metric '%s{%v}' not found", family, expectedLabels)
```

Comment:
the same here:
```suggestion
	return -1, fmt.Errorf("metric '%s{%v}' not found", family, expectedLabels)
```

Reply:
Done in d1ffef3bfd12c67c8135e89acaeedfca3cc0206a